lua/framework/lua_app.cc

```diff
@@ -0,0 +1,262 @@
+#include "lua/framework/lua_app.h"
+#include "framework/logging/log.h"
+#include "lua/framework/console/console.h"
+#include "framework/runtime.h"
+#include "framework/object_factory.h"
+#include "framework/utils/timer.h"
+#include "framework/event_system/event.h"
+#include "framework/event_system/system_wide_event_publisher.h"
+#ifndef NDEBUG
+#include <unistd.h>
+#endif
+#include "petsc.h"
+
+using namespace opensn;
+
+namespace opensnlua
+{
+
+const std::string command_line_help_string_ =
+  "\nUsage: exe inputfile [options values]\n"
+  "\n"
+  "     -v                          Level of verbosity. Default 0.\n"
+  "                                 Can be either 0, 1 or 2.\n"
+  "     a=b                         Executes argument as a lua string. "
+  "i.e. x=2 or y=[[\"string\"]]\n"
+  "     --allow-petsc-error-handler Allows petsc error handler.\n"
+  "     --suppress-beg-end-timelog  Suppresses time logs at the \n"
+  "                                 beginning and end of execution.\n"
+  "     --suppress-color            Suppresses the printing of color.\n"
+  "                                 useful for unit tests requiring a diff.\n"
+  "     --dump-object-registry      Dumps the object registry.\n"
+  "\n\n\n";
+
+LuaApp::LuaApp(MPI_Comm comm)
+{
+  int location_id = 0, number_processes = 1;
+  MPI_Comm_rank(comm, &location_id);
+  MPI_Comm_size(comm, &number_processes);
+
+  opensn::mpi.SetCommunicator(comm);
+  opensn::mpi.SetLocationID(location_id);
+  opensn::mpi.SetProcessCount(number_processes);
+}
+
+LuaApp::~LuaApp()
+{
+}
+
+int
+LuaApp::InitPetSc(int argc, char** argv)
+{
+  PETSC_COMM_WORLD = mpi.comm;
+  PetscOptionsInsertString(nullptr, "-error_output_stderr");
+  if (not allow_petsc_error_handler_) PetscOptionsInsertString(nullptr, "-no_signal_handler");
+
+  PetscCall(PetscInitialize(&argc, &argv, nullptr, nullptr));
+
+  return 0;
+}
+
+int
+LuaApp::Run(int argc, char** argv)
+{
+  InitPetSc(argc, argv);
+  opensn::Initialize();
+  console.PostMPIInfo(opensn::mpi.location_id, opensn::mpi.process_count);
+
+  int error_code = 0;
+  ParseArguments(argc, argv);
+  if (!termination_posted_)
```

Comment:
```suggestion
  if (not termination_posted_)
```

Reply:
I need to put this on a postit note and glue that on my monitor. I forgot about this again 🤦‍♂️